src/llm_phase.py

```
@@ -581,6 +695,51 @@ def _process_batch(self, batch: List[Tuple[str, str]], **kwargs) -> List[str]:
+                        f"Batch processing had {len(failed_indices)} failed responses out of {len(batch_responses)}"
+                    )
+
+                    if not self.enable_retry:
+                        # No retry allowed - fail immediately
+                        first_failed = batch_responses[failed_indices[0]]
+                        failed_content = first_failed.get("content", "Unknown error")
+                        block_info = f"batch index {failed_indices[0]}"
+                        raise GenerationFailedError(
+                            message=f"Batch generation failed (retry disabled): {failed_content[:200]}",
+                            block_info=block_info,
+                        )
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)